src/fastcs/transports/epics/util.py

```
@@ -1,6 +1,49 @@
+) -> None:
+    """Reject controller ids that wouldn't be safe in an EPICS PV name.
 
-def controller_pv_prefix(prefix: str, controller_api: ControllerAPI) -> str:
-    return ":".join([prefix] + [snake_to_pascal(node) for node in controller_api.path])
+    Rejects ids with characters not matched by ``id_re`` and rejects setups
+    where the longest derivable PV prefix already exceeds the
+    ``EPICS_MAX_NAME_LENGTH`` PV name limit. ``transport_label`` is the
+    transport-specific tag (e.g. ``"EPICS CA id"``) that appears in the
+    illegal-characters error message.
+    """
+    id = controller_api.path[0]
```

Ruff (0.15.13)

[error] Variable `id` is shadowing a Python builtin

(A001)